internal/webhooks/nodepool_webhook.go

```diff
@@ -170,6 +171,36 @@ func (webhook *DockyardsNodePool) validate(ctx context.Context, oldNodePool, new
 		}
 	}
 
+	if newNodePool.Labels[dockyardsv1.LabelOrganizationName] != newNodePool.Namespace {
+		invalid := field.Invalid(
+			field.NewPath("metadata",
+				"labels",
+				dockyardsv1.LabelOrganizationName),
+			newNodePool.Labels[dockyardsv1.LabelOrganizationName],
+			fmt.Sprintf("incorrect organization label value, expected '%s'", newNodePool.Namespace))
+		errorList = append(errorList, invalid)
+	}
+
+	ownerReferences := newNodePool.GetOwnerReferences()
+	var clusterOwner string
+	for _, ref := range ownerReferences {
+		if ref.Kind == dockyardsv1.ClusterKind && ref.APIVersion == dockyardsv1.GroupVersion.String() {
+			clusterOwner = ref.Name
+
+			break
+		}
+	}
+
+	if newNodePool.Labels[dockyardsv1.LabelClusterName] != clusterOwner {
```

Comment:
Same thing more or less, I'm not sure how long it will be true that node pools are always owned by clusters. Just verifying existence of the label is good enough.